### PR TITLE
G bhutani12/population balance equation - request to merge the population balance branch into the master branch

### PR DESCRIPTION
**Pull request by gbhutani**
Population balance equation solver using the direct quadrature method of moments (DQMOM). Short and medium tests pass. One test -- check_use_statements -- throws a warning; I have no idea why this is the case.

### CODE REVIEW

**Comment by jrper** [comment]
Apologies for failing to get this queued on buildbot until now. The queue is at 

http://buildbot.ese.ic.ac.uk:8080/builders/population-balance-equation

 In doing so, it appears we've run into a problem in building the manual due to @gbhutani using a couple of packages from texlive-science, which isn't in the vanilla AMCG build, but is evidently on both of our machines. Unless @tmbgreaves tells me otherwise, I will assume we can't just add these, and will try and work out what needs to be changed in the manual to avoid the extra packages.

**Comment by tmbgreaves** [comment]
I think it's perfectly reasonable to add more texlive packages if they're needed to do what's needed in the manual (and are packaged by Ubuntu). Let me know what, and I'll add them in to fluidity-dev.

**Comment by jrper** [comment]
I believe it's just texlive-science, but I don't have a clean enough VM or docker environment anywhere to check. I'm currently attempting to rectify that failing.

**Comment by jrper** [comment]
Confirming that installing texlive-science clears the error on xenial.

**Comment by tmbgreaves** [comment]
Leave that with me - I'll aim to get it added and pushed to the ppa by Wednesday evening at the latest.

**Comment by tmbgreaves** [comment]
The package set has been updated, and the xenial image has been rebuilt so all new buildbot builds should happen with texlive-science present.

**Comment by tmbgreaves** [comment]
One warning from check_use_statements; see:

http://buildbot.ese.ic.ac.uk:8080/builders/population-balance-equation/builds/1/steps/shell_8/logs/stdio

check_use_statements: Assigning variables:
check_use_statements: Assigning out = ../../population_balance/DQMOM.F90:
  use fldebug
  use vector_tools
  use spud
  use global_parameters, only: OPTION_PATH_LEN, FIELD_NAME_LEN
  use futils, only: int2str
  use elements
  use sparse_tools
  use fetools
  use fields
  use state_module
  use state_fields_module
  use initialise_fields_module
  use solvers

check_use_statements: Running warning tests: 
check_use_statements: Running Check_use_statements:
../../population_balance/DQMOM.F90:
  use fldebug
  use vector_tools
  use spud
  use global_parameters, only: OPTION_PATH_LEN, FIELD_NAME_LEN
  use futils, only: int2str
  use elements
  use sparse_tools
  use fetools
  use fields
  use state_module
  use state_fields_module
  use initialise_fields_module
  use solvers

check_use_statements: warning.
check_use_statements: W

**Comment by gbhutani** [comment]
The test passes on my local machine (Ubuntu 14.04). Any ideas why it is
throwing the warning?

*Dr Gaurav Bhutani*
Research Associate
Froth & Foam Research Group
Department of Earth Science and Engineering
2.58 Royal School of Mines
Imperial College London
South Kensington Campus
London SW7 2AZ
Tel: +44(0)20 7594 7144
Email: gaurav.bhutani@gmail.com

**Comment by jrper** [comment]
I assume you've found a bug in my script which generates the reorderings when running on your machine. I can't get it to reoccur though. I'm happy with the code otherwise I think, so if somebody is willing to reorder the modules for DQMOM.F90 then I think this is good to go.

P.S. Is there any chance you fixed this, but didn't add the file when committing the change to git? If you've updated your local copy, but not committed the change, that would explain it.

**Comment by jrper** [comment]
Ah, I think I've found the problem. That the misordering in that file didn't originally appear seems to be my fault, coupled to something to do with 14.04. Since that isn't directly related to this branch, I'll put up a separate pull request once I've worked out a fix.

File: assemble/Diagnostic_fields_wrapper.F90 [hunk 1]
```diff
@@ -555,6 +556,11 @@ subroutine calculate_diagnostic_variables(state, exclude_nonrecalculated)
           call calculate_sediment_active_layer_volume_fractions(state(i))
        end if
        ! End of sediment diagnostics.
+       
+       ! Start of population balance diagnostics.
+          call dqmom_calculate_moments(state(i))
```
Comment (stephankramer) [inline assemble/Diagnostic_fields_wrapper.F90]: Indentation

File: assemble/Multiphase.F90 [hunk 2]
```diff
@@ -397,8 +397,8 @@ subroutine assemble_fluid_particle_drag(istate_fluid, istate_particle)
                   if(have_option(trim(state(istate_particle)%option_path)//"/multiphase_properties/particle_diameter")) then
                      call get_option(trim(state(istate_particle)%option_path)//"/multiphase_properties/particle_diameter", d)
                      have_constant_d = .true.
-                  else if(have_option(trim(state(istate_particle)%option_path)//"/multiphase_properties/particle_diameter_use_scalar_field")) then
-                     call get_option(trim(state(istate_particle)%option_path)//"/multiphase_properties/particle_diameter_use_scalar_field", d_field_name)
+                  else if(have_option(trim(state(istate_particle)%option_path)//"/multiphase_properties/particle_dia_use_scalar_field")) then
+                     call get_option(trim(state(istate_particle)%option_path)//"/multiphase_properties/particle_dia_use_scalar_field", d_field_name)
```
Comment (stephankramer) [inline assemble/Multiphase.F90]: Why was this option renamed? `particle_dia` sounds a bit random to me, and in general we don't use abbreviations in options names

File: main/Fluids.F90 [hunk 3]
```diff
@@ -302,6 +306,9 @@ SUBROUTINE FLUIDS()
        call populate_sub_state(state,sub_state)
     end if
 
+    ! set population balance initial conditions
+       call dqmom_init(state)
```
Comment (stephankramer) [inline main/Fluids.F90]: Indentation

File: main/Fluids.F90 [hunk 4]
```diff
@@ -576,6 +583,9 @@ SUBROUTINE FLUIDS()
 
           call compute_goals(state)
 
+          ! Calculate source terms for population balance scalars
+             call dqmom_calculate_source_terms(state, ITS)
```
Comment (stephankramer) [inline main/Fluids.F90]: Indentation

**Pull request merged**